Shave and libtool-2.2 don't get along

--- origsrc/gnote-0.6.2/configure.ac	2009-08-11 17:02:43.000000000 -0500
+++ src/gnote-0.6.2/configure.ac	2009-11-17 23:26:48.027975400 -0600
@@ -4,6 +4,8 @@ AC_CONFIG_SRCDIR(README)
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 AM_MAINTAINER_MODE
 
+m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
+
 GNOME_COMMON_INIT
 GNOME_DOC_INIT
 
@@ -209,12 +211,8 @@ AC_SUBST(GNOTE_LOCALEDIR)
 
 AC_DEFINE_UNQUOTED([GNOTE_BUILD_CONFIG], ["$GNOTE_BUILD_CONFIG"], [The string used to hardcode the build config.])
 
-SHAVE_INIT
-
 AC_CONFIG_FILES([
 Makefile
-shave
-shave-libtool
 data/Makefile
 data/gnote.desktop.in
 data/icons/Makefile
