--- origsrc/gnote-0.7.3/src/Makefile.am	2010-11-04 08:19:09.000000000 -0500
+++ src/gnote-0.7.3/src/Makefile.am	2010-11-04 13:24:50.000000000 -0500
@@ -1,6 +1,6 @@
 
 
-SUBDIRS = addins
+SUBDIRS = dbus . addins
 
 EXTRA_DIST = gnote-applet.in
 
@@ -78,7 +78,6 @@ endif
 
 
 if HAVE_DBUS
-SUBDIRS += dbus
 DBUS_SOURCES=remotecontrolproxy.hpp remotecontrolproxy.cpp \
 	dbus/remotecontrol.hpp dbus/remotecontrol.cpp \
 	dbus/remotecontrolclient.hpp dbus/remotecontrolclient.cpp \
@@ -163,3 +162,5 @@ gnote_LDADD = \
 	$(GNOTE_LIBS) \
 	-lX11 \
 	$(NULL)
+
+gnote_LDFLAGS = -Wl,--export-all-symbols,--out-implib,libgnote.exe.a
--- origsrc/gnote-0.7.3/src/addins/addins.mk.in	2010-09-26 03:44:19.000000000 -0500
+++ src/gnote-0.7.3/src/addins/addins.mk.in	2010-11-04 13:24:50.000000000 -0500
@@ -7,6 +7,9 @@ AM_CPPFLAGS=@LIBGTKMM_CFLAGS@ @LIBGLIBMM
 	@GCONF_CFLAGS@ @LIBXML_CFLAGS@ \
 	-DDATADIR=\"$(datadir)\" -DLIBDIR=\"$(libdir)\"
 
-AM_LDFLAGS = -avoid-version -module -export-dynamic
+AM_LDFLAGS = -avoid-version -module -export-dynamic -no-undefined \
+	-Wl,$(top_builddir)/src/libgnote.exe.a \
+	$(BOOST_FILESYSTEM_LIBS) $(BOOST_SYSTEM_LIBS) \
+	$(LIBGTKMM_LIBS) $(GCONF_LIBS) $(LIBXML_LIBS) $(PCRE_LIBS)
 
 ADDINSDIR = @libdir@/@PACKAGE_NAME@/addins/@VERSION@
--- origsrc/gnote-0.7.3/src/dbus/Makefile.am	2010-09-26 03:44:19.000000000 -0500
+++ src/gnote-0.7.3/src/dbus/Makefile.am	2010-11-04 13:24:50.000000000 -0500
@@ -6,8 +6,10 @@ remotecontrol-glue.hpp: gnote-introspect
 remotecontrol-client-glue.hpp: gnote-introspect.xml
 	dbusxx-xml2cpp $^ --proxy=$@
 
+if HAVE_DBUS
 BUILT_SOURCES = remotecontrol-glue.hpp remotecontrol-client-glue.hpp
 CLEANFILES = $(BUILT_SOURCES)
+endif
 
 dist-hook:
 	cd $(distdir); rm -f $(BUILT_SOURCES)
