http://bugzilla.gnome.org/show_bug.cgi?id=603925

--- origsrc/gnote-0.7.1/m4/boost.m4	2010-01-03 09:30:23.000000000 -0600
+++ src/gnote-0.7.1/m4/boost.m4	2010-01-30 22:12:05.393007200 -0600
@@ -59,7 +59,7 @@ dnl Ultrix and Pyramid sh refuse to redi
 (eval "$ac_cpp conftest.$ac_ext") 2>&AS_MESSAGE_LOG_FD |
   $SED -n -e "$1" >conftest.i 2>&1],
   [$3],
-  [$4])dnl
+  [$4])
 rm -f conftest*
 ])# AC_EGREP_CPP
 
