--- origsrc/gnote-0.8.0/src/Makefile.am	2011-09-24 10:40:37.000000000 -0500
+++ src/gnote-0.8.0/src/Makefile.am	2011-09-25 22:11:49.670718600 -0500
@@ -1,6 +1,6 @@
 
 
-SUBDIRS = addins
+SUBDIRS = dbus . addins
 
 EXTRA_DIST = gnote-applet.in
 
@@ -80,7 +80,6 @@ CLEANFILES = gnote-applet
 endif
 
 
-SUBDIRS += dbus
 DBUS_SOURCES=remotecontrolproxy.hpp remotecontrolproxy.cpp \
 	dbus/remotecontrol.hpp dbus/remotecontrol.cpp \
 	dbus/remotecontrolclient.hpp dbus/remotecontrolclient.cpp \
@@ -166,3 +165,5 @@ gnote_LDADD = \
 	$(GNOTE_LIBS) \
 	-lX11 \
 	$(NULL)
+
+gnote_LDFLAGS = $(AM_LDFLAGS) -Wl,--out-implib,libgnote.exe.a
--- origsrc/gnote-0.8.0/src/addins/addins.mk.in	2011-09-24 10:40:37.000000000 -0500
+++ src/gnote-0.8.0/src/addins/addins.mk.in	2011-09-25 21:51:32.382157200 -0500
@@ -7,6 +7,8 @@ AM_CPPFLAGS=@LIBGTKMM_CFLAGS@ @LIBGLIBMM
 	@LIBXML_CFLAGS@ \
 	-DDATADIR=\"$(datadir)\" -DLIBDIR=\"$(libdir)\"
 
-AM_LDFLAGS = -avoid-version -module -export-dynamic
+AM_LDFLAGS = -avoid-version -module -export-dynamic -no-undefined \
+	-Wl,$(top_builddir)/src/libgnote.exe.a \
+	$(LIBGTKMM_LIBS) $(LIBGLIBMM_LIBS) $(LIBXML_LIBS) $(PCRE_LIBS)
 
 ADDINSDIR = @libdir@/@PACKAGE_NAME@/addins/@VERSION@
