--- origsrc/gnote-3.10.3/src/gnome_keyring/ring.cpp	2014-01-10 17:12:14.000000000 -0600
+++ src/gnote-3.10.3/src/gnome_keyring/ring.cpp	2014-03-24 23:14:54.225786900 -0500
@@ -100,7 +100,7 @@ void Ring::clear_password(const std::map
 
 GHashTable *Ring::keyring_attributes(const std::map<std::string, std::string> & atts)
 {
-  GHashTable *result = g_hash_table_new_full(g_str_hash, g_str_equal, free, free);
+  GHashTable *result = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
   for(std::map<std::string, std::string>::const_iterator iter = atts.begin();
       iter != atts.end(); ++iter) {
     g_hash_table_insert(result, strdup(iter->first.c_str()), strdup(iter->second.c_str()));
