--- origsrc/gnote-0.5.3/src/Makefile.am	2009-07-01 12:41:09.000000000 -0500
+++ src/gnote-0.5.3/src/Makefile.am	2009-07-19 10:36:37.813192500 -0500
@@ -1,6 +1,6 @@
 
 
-SUBDIRS = addins
+SUBDIRS = . addins
 
 EXTRA_DIST = gnote-applet.in
 
@@ -13,16 +13,18 @@ AM_CPPFLAGS=@LIBGTKMM_CFLAGS@ @LIBGLIBMM
 	-DGNOTE_LOCALEDIR=\"@GNOTE_LOCALEDIR@\" \
 	-DDATADIR=\"$(datadir)\" -DLIBDIR=\"$(libdir)\"
 
-AM_LDFLAGS=-Wl,--export-dynamic
+AM_LDFLAGS=-export-dynamic
 
-GNOTE_LIBS = libgnote.a @LIBGLIBMM_LIBS@ @LIBGTKMM_LIBS@ \
-	@GCONF_LIBS@ @BOOST_FILESYSTEM_LIBS@ \
+GNOTE_LIBS = \
+	$(top_builddir)/libtomboy/libtomboy.la \
+	@LIBGLIBMM_LIBS@ @LIBGTKMM_LIBS@ \
+	@GCONF_LIBS@ @BOOST_FILESYSTEM_LIBS@ @BOOST_SYSTEM_LIBS@ \
 	@LIBXSLT_LIBS@ \
 	@PCRE_LIBS@ \
 	@GTKSPELL_LIBS@ @GTK_LIBS@ \
-	-luuid -L$(top_builddir)/libtomboy -ltomboy
+	-luuid
 
-noinst_LIBRARIES = libgnote.a
+lib_LTLIBRARIES = libgnote.la
 bin_PROGRAMS = gnote
 check_PROGRAMS = trietest stringtest notetest dttest uritest filestest \
 	xmlreadertest
@@ -32,26 +34,32 @@ TESTS = trietest stringtest notetest dtt
 
 trietest_SOURCES = test/trietest.cpp \
 	sharp/string.cpp debug.cpp
+trietest_CXXFLAGS =
 trietest_LDADD =  @PCRE_LIBS@ @LIBGLIBMM_LIBS@
 
 dttest_SOURCES = test/dttest.cpp \
 	sharp/datetime.cpp debug.cpp
+dttest_CXXFLAGS =
 dttest_LDADD = @LIBGLIBMM_LIBS@
 
 stringtest_SOURCES = test/stringtest.cpp \
 	sharp/string.cpp debug.cpp
+stringtest_CXXFLAGS =
 stringtest_LDADD =  @PCRE_LIBS@ @LIBGLIBMM_LIBS@
 
 filestest_SOURCES = test/filestest.cpp \
 	sharp/files.cpp
+filestest_CXXFLAGS =
 filestest_LDADD = @BOOST_FILESYSTEM_LIBS@
 
 uritest_SOURCES = test/uritest.cpp \
 	sharp/string.cpp  sharp/uri.cpp debug.cpp
+uritest_CXXFLAGS =
 uritest_LDADD =  @PCRE_LIBS@ @LIBGLIBMM_LIBS@
 
 xmlreadertest_SOURCES = test/xmlreadertest.cpp \
 	sharp/xmlreader.cpp debug.cpp
+xmlreadertest_CXXFLAGS =
 xmlreadertest_LDADD = @LIBXML_LIBS@
 
 notetest_SOURCES = test/notetest.cpp
@@ -73,7 +81,7 @@ gnote-applet: gnote-applet.in Makefile
 CLEANFILES = gnote-applet
 endif
 
-libgnote_a_SOURCES = \
+libgnote_la_SOURCES = \
 	base/singleton.hpp \
 	base/macros.hpp \
 	base/inifile.hpp base/inifile.cpp \
@@ -142,5 +150,9 @@ libgnote_a_SOURCES = \
 	$(APPLET_SOURCES)
 	$(NULL)
 
-gnote_LDADD = $(GNOTE_LIBS)
+libgnote_la_LDFLAGS = -no-undefined
+
+libgnote_la_LIBADD = $(GNOTE_LIBS)
+
+gnote_LDADD = libgnote.la
 
--- origsrc/gnote-0.5.3/src/addins/addins.mk.in	2009-05-27 18:05:24.000000000 -0500
+++ src/gnote-0.5.3/src/addins/addins.mk.in	2009-07-19 10:18:00.414582400 -0500
@@ -7,6 +7,8 @@ AM_CPPFLAGS=@LIBGTKMM_CFLAGS@ @LIBGLIBMM
 	@GCONF_CFLAGS@ @LIBXML_CFLAGS@ \
 	-DDATADIR=\"$(datadir)\" -DLIBDIR=\"$(libdir)\"
 
-AM_LDFLAGS = -avoid-version -module -Wl,--export-dynamic
+AM_LDFLAGS = -avoid-version -module -no-undefined -export-dynamic
+
+LIBS = $(top_builddir)/src/libgnote.la @LIBGTKMM_LIBS@
 
 ADDINSDIR = @libdir@/@PACKAGE_NAME@/addins/@VERSION@
